File: src/main/java/com/nickcoblentz/montoya/settings/ExtensionSettingsFormGenerator.java
package com.nickcoblentz.montoya.settings;

import de.milchreis.uibooster.UiBooster;
import de.milchreis.uibooster.model.*;

import java.util.List;

public class ExtensionSettingsFormGenerator {
    private final String FormName;
    List<IExtensionSetting> ExtensionSettings;
    private FormBuilder SettingsFormBuilder;

    public ExtensionSettingsFormGenerator(List<IExtensionSetting> extensionSettings, String formName) {
        ExtensionSettings = extensionSettings;
        FormName=formName;
        buildForm();
    }

    private void buildForm()
    {
        UiBooster booster = new UiBooster(UiBoosterOptions.Theme.DEFAULT);
        SettingsFormBuilder = booster.createForm(FormName)
            .setChangeListener((element, value, form) -> {
                form.getById("Change").setValue("Settings have not yet been saved!");
            });

        ExtensionSettings.forEach((s) -> {
            SettingsFormBuilder
                    .startRow()
                    .addText(s.getName(),s.getCurrentValue()).setID(s.getStorageKey())
                    /*.addButton("\u27F3", new ButtonClickListener() {
                        @Override
                        public void onClick(FormElement formElement, Form form) {
                            form.getById(s.getStorageKey()).setValue(s.reset());
                        }
                    }).setMargin(25,0,0,0)*/
                    .endRow();
                });

            SettingsFormBuilder
                .startRow().addLabel(" ").setID("Change").setMargin(0,15,0,5).endRow()
                .startRow()
                .addButton("Reset", new ButtonClickListener() {
                    @Override
                    public void onClick(FormElement formElement, Form form) {
                        ExtensionSettings.forEach((item) ->{
                            form.getById(item.getStorageKey()).setValue(item.reset());
                        });
                    }
                })
                .addButton("Save", new ButtonClickListener() {
                    @Override
                    public void onClick(FormElement formElement, Form form) {
                        ExtensionSettings.forEach((item) ->{
                            if(form.getById(item.getStorageKey()).getValue() instanceof String sValue)
                            {
                                item.setCurrentValue(sValue);
                                item.save();
                            }
                        });
                        form.getById("Change").setValue(" ");
                    }
                })
                .addButton("Close", new ButtonClickListener() {
                    @Override
                    public void onClick(FormElement formElement, Form form) {
                        form.close();
                    }
                })
            .endRow();
    }

    public FormBuilder getSettingsFormBuilder() {
        return SettingsFormBuilder;
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/ExtensionSettingsFormGenerator.kt
package com.nickcoblentz.montoya.settings

import de.milchreis.uibooster.UiBooster
import de.milchreis.uibooster.model.*
import java.util.function.Consumer

class ExtensionSettingsFormGenerator(var extensionSettings: List<IExtensionSetting>, private val formName: String) {
    val booster = UiBooster(UiBoosterOptions.Theme.DEFAULT)
    var settingsFormBuilder: FormBuilder = booster.createForm(formName)
        private set

    init {
        buildForm()
    }

    private fun buildForm() {

        settingsFormBuilder.setChangeListener { element: FormElement<*>?, value: Any?, form: Form ->
                form.getById("Change").setValue("Settings have not yet been saved!")
            }

        extensionSettings.forEach(Consumer { s: IExtensionSetting ->
            settingsFormBuilder
                .startRow()
                .addText(s.name, s.currentValue).setID(s.storageKey) /*.addButton("\u27F3", new ButtonClickListener() {
                    @Override
                    public void onClick(FormElement formElement, Form form) {
                        form.getById(s.getStorageKey()).setValue(s.reset());
                    }
                }).setMargin(25,0,0,0)*/
                .endRow()
        })

        settingsFormBuilder
            .startRow().addLabel(" ").setID("Change").setMargin(0, 15, 0, 5).endRow()
            .startRow()
            .addButton(
                "Reset"
            ) { formElement: FormElement<*>?, form: Form ->
                extensionSettings.forEach(
                    Consumer { item: IExtensionSetting ->
                        form.getById(item.storageKey).setValue(item.reset())
                    })
            }
            .addButton("Save") { formElement: FormElement<*>?, form: Form ->
                extensionSettings.forEach(
                    Consumer<IExtensionSetting> { item: IExtensionSetting ->
                        (form.getById(item.storageKey).value as? String)?.let {
                            item.currentValue = it
                            item.save()
                        }
                    })
                form.getById("Change").setValue(" ")
            }
            .addButton(
                "Close"
            ) { formElement: FormElement<*>?, form: Form -> form.close() }
            .endRow()
    }
}